frontend/src/metabase/search/containers/SearchApp.jsx

```diff
@@ -25,172 +30,90 @@ import {
   SearchRoot,
 } from "./SearchApp.styled";
 
-const PAGE_SIZE = 50;
-
-const SEARCH_FILTERS = [
-  {
-    name: t`Apps`,
-    filter: "app",
-    icon: "star",
-  },
-  {
-    name: t`Dashboards`,
-    filter: "dashboard",
-    icon: "dashboard",
-  },
-  {
-    name: t`Collections`,
-    filter: "collection",
-    icon: "folder",
-  },
-  {
-    name: t`Databases`,
-    filter: "database",
-    icon: "database",
-  },
-  {
-    name: t`Models`,
-    filter: "dataset",
-    icon: "model",
-  },
-  {
-    name: t`Raw Tables`,
-    filter: "table",
-    icon: "table",
-  },
-  {
-    name: t`Questions`,
-    filter: "card",
-    icon: "bar",
-  },
-  {
-    name: t`Pulses`,
-    filter: "pulse",
-    icon: "pulse",
-  },
-  {
-    name: t`Metrics`,
-    filter: "metric",
-    icon: "sum",
-  },
-  {
-    name: t`Segments`,
-    filter: "segment",
-    icon: "segment",
-  },
-];
-
 export default function SearchApp({ location }) {
   const { handleNextPage, handlePreviousPage, setPage, page } = usePagination();
-  const [filter, setFilter] = useState(location.query.type);
 
-  const handleFilterChange = filterItem => {
-    setFilter(filterItem && filterItem.filter);
-    setPage(0);
-  };
+  const searchText = useMemo(
+    () => getSearchTextFromLocation(location),
+    [location],
+  );
 
-  const query = {
-    q: location.query.q,
-    limit: PAGE_SIZE,
-    offset: PAGE_SIZE * page,
-  };
+  const searchFilters = useMemo(() => {
+    return getFiltersFromLocation(location);
+  }, [location]);
+
+  const [selectedSidebarType, setSelectedSidebarType] = useState(null);
+
+  useEffect(() => {
+    setSelectedSidebarType(null);
+  }, [searchText, searchFilters]);
```

Comment:
Yep. Basically, we want to make sure `All results` is selected when we change the search from the `SearchApp` (i.e. when we use the `SearchBar` to change filters or text). For example, if the user was already on the search page and clicked `Questions` on the sidebar then ran another search, then the sidebar will reset. I thought about putting this logic in the `TypeSearchSidebar`, but that felt super hidden considering that we're using the `sidebarType` as a filter in `query`.

Reply:
One problem with this kind of problem is that it's not obvious to readers why it is in the dependency. I initially thought maybe this has gone through different iterations and `searchFilters` used to be in the `useEffect` body.

One thing you can make this obvious is to use `searchFilters` in the body.

```ts
useEffect(() => {
    if(searchFilters) {
      setSelectedSidebarType(null);
    }
  }, [searchText, searchFilters]);
```

But please make sure `searchFilters` won't be null otherwise, it won't call `setSelectedSidebarType(null)`

Reply:
Would it make more sense we used `location.search` or something to that effect? Since the search changes with the URL, then we just use a single string dependency

Reply:
That could work too, but I'd still put that in the `useEffect` body just to make sure no one will mistakenly remove it from the dependency array. If you put `if (location.search)` similar to my previous suggestion, removing `location.search` from the dependency array would create an ESLint warning that you should put it back in the dependency array.

Reply:
Ah I get what you mean. I'll change that